Lib/asyncio/base_events.py

```diff
@@ -555,10 +560,22 @@ def _check_default_executor(self):
         if self._executor_shutdown_called:
             raise RuntimeError('Executor shutdown has been called')
 
+    async def _asyncgen_close(self, agen):
+        await agen.aclose()
```

Comment:
this is just side-stepping the actual bug by delaying the creation of the async_generator_aclose object to avoid the incorrect warning